src/pages/QuestionPage/index.jsx

```diff
@@ -0,0 +1,144 @@
+    return () => {
+      clearInterval(timer);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!quizInfo.loading && quizInfo.data.length !== 0) {
+      setActiveQuiz(quizInfo.data);
+    }
+  }, [quizInfo]);
+  useEffect(() => {
+    if (timeLeft === 0) {
+      if (activeQuestionNo < questions.data.length - 1) {
+        setOptionColor("light");
+        setActiveOption(-1);
+        setActiveQuestionNo(activeQuestionNo + 1);
+      }
+      clearInterval(timer);
+    }
+  }, [timeLeft]);
+  return (
+    <Layout>
+      <section className="question-section w-5-6 text-light mt-5">
+        {questions.loading && <Loader />}
+        {!questions.loading && questions.data.length > 0 && (
+          <>
+            <h1 className="text-3xl text-bold mb-4 text-center text-primary">
+              {activeQuiz.title}
+            </h1>
+            <div className="col gap-2">
+              <div className="row justify-between items-center">
+                <h3 className="text-xl ">
+                  Question:{" "}
+                  <span className="text-medium  text-primary ml-05">
+                    {activeQuestionNo + 1}/{questions.data.length}
+                  </span>
+                </h3>
+                <div className="row gap-1">
+                  <h3 className="text-xl ">
+                    Time Left:
+                    <span className="text-medium  text-primary ml-05">
+                      {timeLeft} Seconds
+                    </span>
+                  </h3>
+                  <h3 className="text-xl ">
+                    Score:
+                    <span className="text-medium  text-primary ml-05">
+                      {score}
+                    </span>
+                  </h3>
+                </div>
+              </div>
+              <div className="col gap-1">
+                <h3 className="text-light text-medium font-normal">
+                  {questions.data[activeQuestionNo].title}
+                </h3>
+                <div className="col options gap-1 ">
+                  {questions.data[activeQuestionNo].options.map((option, i) => (
+                    <button
+                      className={`btn  ${
+                        activeOption === i && !option.isCorrect
+                          ? "btn-error"
+                          : ""
+                      }  btn-${option.isCorrect ? optionStateColor : "light"} 
+                     `}
+                      key={option.value}
+                      onClick={() => {
+                        setOptionColor("success");
+                        if (option.isCorrect) {
+                          if (activeQuiz.quizDifficulty === "Easy") {
+                            setScore(score + 5);
+                          } else if (activeQuiz.quizDifficulty === "Medium") {
+                            setScore(score + 10);
+                          } else if (activeQuiz.quizDifficulty === "Hard") {
+                            setScore(score + 15);
```

Comment:
You can move this logic in a separate function and maybe switch instead of deeply nested if else

Reply:
I will do that